commons/src/main/scala/dagr/commons/reflect/ReflectionUtil.scala

```diff
@@ -290,7 +292,11 @@ object ReflectionUtil {
     */
   private[reflect] def typedValueFromString(resultType: Class[_], unitType: Class[_], value: String*): Try[Any] = Try {
     if (ReflectionUtil.isCollectionClass(resultType)) {
-      val typedValues = value.map(v => buildUnitFromString(unitType, v).get).asInstanceOf[Seq[java.lang.Object]]
+      val typedValues = value.indexOf(SpecialEmptyOrNoneToken) match {
```

Comment:
This wont work since the default values are passed into this method along with `:None:` and the values that should overwrite the default values.